src/context/IIIFResourceContext.tsx

```diff
@@ -45,39 +130,102 @@ export function IIIFContentProvider(props: {
+        disablePaging: false,
+      });
 
-    if (typeof props.resource === 'string') {
-      const controller = new AbortController();
+      dispatch({
+        type: 'updateSequences',
+        sequences: resourceSequences,
+      });
 
-      fetch(props.resource, { signal: controller.signal })
-        .then((json) => {
-          setResource(json as any);
-          if (onLoad) {
-            onLoad(json);
-          }
-        })
-        .catch((e) => setError(e));
+      if (isControlled) {
+        const id = resource.items[0].id;
 
-      return () => {
-        controller.abort();
-      };
-    } else {
-      setResource(props.resource);
-      if (onLoad) {
-        onLoad(props.resource);
+        // Pass back the first resource id on load
+        onResourceChanged && onResourceChanged(id);
+
+        // If current resource id is controlled, update context
+        dispatch({
+          type: 'updateCurrentId',
+          id,
+        });
       }
     }
-  }, [props.resource]);
+  }, [resource]);
+
+  useEffect(() => {
+    if (currentResourceId && onResourceChanged) {
+      onResourceChanged(currentResourceId);
+    }
+  }, [currentResourceId]);
+
+  const next = () => {
+    //const nextResourceId = getNextResourceItemId(currentResourceId, resource, sequences);
 
-  const value = useMemo(() => {
-    return { resource: mergedResource, error, isLoaded: !!resource, orientation, currentResourceId };
-  }, [mergedResource, error, orientation, currentResourceId]);
+    const nextResourceId = getNavResourceItemId({
+      currentResourceId,
+      direction: 'next',
+      resource,
+      sequences,
+    });
 
-  return <ReactContext.Provider value={value}>{props.children}</ReactContext.Provider>;
+    nextResourceId &&
+      dispatch({
+        type: 'updateCurrentId',
+        id: nextResourceId,
+      });
+  };
+
+  const prev = () => {
+    const prevResourceId = getNavResourceItemId({
+      currentResourceId,
+      direction: 'prev',
+      resource,
+      sequences,
+    });
+    prevResourceId &&
+      dispatch({
+        type: 'updateCurrentId',
+        id: prevResourceId,
+      });
+  };
+
+  const value = {
+    state: {
+      ...state,
+      isLoaded: !!resource,
+      isEnd: isLastResourceItem(state.currentResourceId, state.resource),
+      isStart: isFirstResourceItem(state.currentResourceId, state.resource),
+      resource: mergedResource,
+    },
+    dispatch,
+    next: {
+      resourceId: getNavResourceItemId({
+        currentResourceId,
+        direction: 'next',
+        resource,
+        sequences,
+      }),
+      handleNextClick: next,
+    },
+    prev: {
+      resourceId: getNavResourceItemId({
+        currentResourceId,
+        direction: 'prev',
+        resource,
+        sequences,
+      }),
+      handlePrevClick: prev,
+    },
+  };
+
+  return <ReactContext.Provider value={value}>{children}</ReactContext.Provider>;
```

Comment:
We should maybe look at splitting the context into actions / state to avoid re-rendering too often. This might be considerable for large manifests with many thumbnails. We could wrap the actions in a `useMemo()` and they should never change (and components using them, never re-render). 

The state could be one or more contexts, but we could measure to see if we need to go that far. 

I don't think this is required at the moment, but something we can look at later.